scripts/run_frontend_tests.py

```diff
@@ -89,14 +89,21 @@ def main(args=None):
             'start', os.path.join('core', 'tests', 'karma.conf.ts')]
 
     task = subprocess.Popen(cmd, stdout=subprocess.PIPE)
-    out, _ = task.communicate()
-    task.wait()
+    output_lines = []
+    # Reads and prints realtime output from the subprocess until it terminates.
+    while True:
+        line = task.stdout.readline()
+        line = python_utils.UNICODE(line, 'utf-8')
+        if len(line) == 0 and task.poll() is not None:
```

Comment:
I got tripped up slightly on the ``is not None`` check. I suspect the subprocess API isn't well known by many Python developers, so I suggest adding a brief comment here just describing what this conditional means (e.g. "If there's no more output and the process has ended, break the loop" or something similar).

Reply:
Done.